### PR TITLE
Pass DC's compile/compile_ultra arguments

### PR DESCRIPTION
**Pull request by donggyukim**
We sometimes want to use various DC's compile/compile_ultra arguments.

### CODE REVIEW

File: src/addons/synthesis/dc/tools/run-synthesis [hunk 1]
```diff
@@ -94,10 +96,8 @@ sed 's/^set DCRM_CONSTRAINTS_INPUT_FILE .*$//' -i $run_dir/rm_setup/dc_setup_fil
 # I don't want to write ICC2 files
 sed 's/^write_icc2_files .*$//' -i $run_dir/rm_dc_scripts/dc.tcl
 
-# FIXME: We aren't licensed for "DC-Extension" so we can't pass "-congestion"
-# or "-spg" to compile_ultra
-find $run_dir -iname "*.tcl" | xargs sed s/'^compile_ultra \(.*\)-congestion\(.*\)'/'compile_ultra \1 \2'/g -i
-find $run_dir -iname "*.tcl" | xargs sed s/'^compile_ultra \(.*\)-spg\(.*\)'/'compile_ultra \1 \2'/g -i
+# Flush compile_ultra arguments and use user specified arguments instead
+find $run_dir -iname "*.tcl" | xargs sed s/'^compile_ultra \(.*\)'/"compile_ultra ${compile_args[*]}"/g -i
```
Comment (colinschmidt) [inline src/addons/synthesis/dc/tools/run-synthesis]: Are the only compile_ultra arguments in the default scripts "-congestion", "-spg" and "-gate_clock". If not it seems like you are no longer passing in the same set of arguments.

Comment (donggyukim) [inline src/addons/synthesis/dc/tools/run-synthesis]: Yes, they are. There were only -gate_clock as a result.

**Comment by colinschmidt** [comment]
LGTM.

**Comment by palmer-dabbelt** [comment]
Thanks!

**Pull request merged**